From 27822a79236daa5f7add97f74c13e282a43349b5 Mon Sep 17 00:00:00 2001
From: zhangmeng <zhangmeng.kevin@spacemit.com>
Date: Mon, 24 Jun 2024 09:22:29 +0800
Subject: [PATCH 0768/1222] riscv: show reason of unaligned access speed are
 different among cores

the debug info is:
cpu6: Ratio of byte access time to unaligned word access is 1.95, unaligned accesses are fast
cpu2: Ratio of byte access time to unaligned word access is 3.03, unaligned accesses are fast
cpu3: Ratio of byte access time to unaligned word access is 2.95, unaligned accesses are fast
cpu4: Ratio of byte access time to unaligned word access is 1.97, unaligned accesses are fast
cpu7: Ratio of byte access time to unaligned word access is 1.95, unaligned accesses are fast
cpu1: Ratio of byte access time to unaligned word access is 2.74, unaligned accesses are fast
cpu5: Ratio of byte access time to unaligned word access is 2.16, unaligned accesses are fast
cpu0: Ratio of byte access time to unaligned word access is 10.16, unaligned accesses are fast

cpu0 is more efficient than other cores because of system bandwith preemption.

Change-Id: I8890cb2657e26bcc692fb689801d870ceb402a9c
---
 arch/riscv/kernel/unaligned_access_speed.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/arch/riscv/kernel/unaligned_access_speed.c b/arch/riscv/kernel/unaligned_access_speed.c
index ae2068425fbcd..839daa4e50d65 100644
--- a/arch/riscv/kernel/unaligned_access_speed.c
+++ b/arch/riscv/kernel/unaligned_access_speed.c
@@ -315,6 +315,15 @@ static void check_vector_unaligned_access(struct work_struct *work __always_unus
 	while ((now = jiffies) == start_jiffies)
 		cpu_relax();
 
+	/*
+	 * show the reason why cpu0 unaligned access is more efficient
+	 * than nonboot cores, the nonboot cores unalgined access is measured
+	 * concurrently.
+	 */
+	pr_info("The real ratio of byte access time to unaligned word access should refer to the value of CPU0\n");
+	pr_info("Cpu0 unaligned access is more efficient than nonboot cores, because of system bandwidth preemption.\n");
+	pr_info("Nonboot cpus' unaligned access ratio measured simultaneously, but cpu0's measure is separately\n");
+
 	/*
 	 * For a fixed amount of time, repeatedly try the function, and take
 	 * the best time in cycles as the measurement.
-- 
2.51.0

